x/stakeibc/keeper/hooks.go

```diff
@@ -184,46 +184,42 @@ func (k Keeper) UpdateRedemptionRates(ctx sdk.Context, depositRecords []recordst
-		return ((record.Status == recordstypes.DepositRecord_DELEGATION_QUEUE ||
-			record.Status == recordstypes.DepositRecord_DELEGATION_IN_PROGRESS) &&
-			record.HostZoneId == hostZone.ChainId)
-	})
-
-	// sum the amounts of the deposit records
+// Determine the deposit account balance, representing native tokens that have been deposited
+// from liquid stakes, but have not yet been transferred to the host
+func (k Keeper) GetDepositAccountBalance(chainId string, depositRecords []recordstypes.DepositRecord) sdk.Dec {
+	// sum on deposit records with status TRANSFER_QUEUE or TRANSFER_IN_PROGRESS
 	totalAmount := sdkmath.ZeroInt()
-	for _, depositRecord := range undelegatedDepositRecords {
-		totalAmount = totalAmount.Add(depositRecord.Amount)
+	for _, depositRecord := range depositRecords {
+		transferStatus := (depositRecord.Status == recordstypes.DepositRecord_TRANSFER_QUEUE ||
+			depositRecord.Status == recordstypes.DepositRecord_TRANSFER_IN_PROGRESS)
+
+		if depositRecord.HostZoneId == chainId && transferStatus {
+			totalAmount = totalAmount.Add(depositRecord.Amount)
+		}
 	}
 
 	return sdk.NewDecFromInt(totalAmount)
 }
 
-// Determine the deposit account balance, representing native tokens that have been deposited
-// from liquid stakes, but have not yet been transferred to the host
-func (k Keeper) GetDepositAccountBalance(hostZone types.HostZone, depositRecords []recordstypes.DepositRecord) sdk.Dec {
-	// filter to only the deposit records for the host zone with status DELEGATION
-	depositAccountRecords := utils.FilterDepositRecords(depositRecords, func(record recordstypes.DepositRecord) (condition bool) {
-		return (record.Status == recordstypes.DepositRecord_TRANSFER_QUEUE ||
-			record.Status == recordstypes.DepositRecord_TRANSFER_IN_PROGRESS) &&
-			record.HostZoneId == hostZone.ChainId
-	})
-
-	// sum the amounts of the deposit records
+// Determine the undelegated balance from the deposit records queued for staking
+func (k Keeper) GetUndelegatedBalance(chainId string, depositRecords []recordstypes.DepositRecord) sdk.Dec {
+	// sum on deposit records with status DELEGATION_QUEUE or DELEGATION_IN_PROGRESS
 	totalAmount := sdkmath.ZeroInt()
-	for _, depositRecord := range depositAccountRecords {
-		totalAmount = totalAmount.Add(depositRecord.Amount)
+	for _, depositRecord := range depositRecords {
+		delegationStatus := (depositRecord.Status == recordstypes.DepositRecord_DELEGATION_QUEUE ||
+			depositRecord.Status == recordstypes.DepositRecord_DELEGATION_IN_PROGRESS)
+
+		if depositRecord.HostZoneId == chainId && delegationStatus {
+			totalAmount = totalAmount.Add(depositRecord.Amount)
+		}
 	}
 
 	return sdk.NewDecFromInt(totalAmount)
 }
 
 // Returns the total delegated balance that's stored in LSM tokens
 // These are identified by any status besides "DEPOSIT_PENDING"
-func (k Keeper) GetLSMDelegations(ctx sdk.Context, chainId string) sdk.Dec {
+func (k Keeper) GetTokenizedDelegation(ctx sdk.Context, chainId string) sdk.Dec {
```

Comment:
GetDelegatedTokenizedTotal
Most other places you use Delegated or Undelegated as an adjective -- just have two adjectives.  This name makes it sound like there is one big delegation and that you are about to return that delegation when really you are about to sum over the many LSM delegations

Reply:
renaming to `GetTotalTokenizedDelegations` to align with host zones

Reply:
going to handle this in the other PR